src/Microsoft.AspNetCore.Razor.Tools/GenerateCommand.cs

```diff
@@ -162,37 +188,43 @@ private OutputItem[] GenerateCode(RazorTemplateEngine templateEngine, IReadOnlyL
         private struct OutputItem
         {
             public OutputItem(
-                SourceItem viewFileInfo,
+                InputItem inputItem,
                 RazorCSharpDocument cSharpDocument)
             {
-                ViewFileInfo = viewFileInfo;
+                InputItem = inputItem;
                 CSharpDocument = cSharpDocument;
             }
 
-            public SourceItem ViewFileInfo { get; }
+            public InputItem InputItem { get; }
 
             public RazorCSharpDocument CSharpDocument { get; }
         }
 
-        private struct SourceItem
+        private struct InputItem
         {
-            public SourceItem(string fullPath, string viewEnginePath)
+            public InputItem(string sourcePath, string outputPath, string physicalRelativePath)
             {
-                FullPath = fullPath;
-                ViewEnginePath = viewEnginePath;
+                SourcePath = sourcePath;
+                OutputPath = outputPath;
+                PhysicalRelativePath = physicalRelativePath;
+                FilePath = '/' + physicalRelativePath.Replace(Path.DirectorySeparatorChar, '/');
             }
 
-            public string FullPath { get; }
+            public string SourcePath { get; }
+
+            public string OutputPath { get; }
+
+            public string PhysicalRelativePath { get; }
 
-            public string ViewEnginePath { get; }
+            public string FilePath { get; }
 
             public Stream CreateReadStream()
             {
                 // We are setting buffer size to 1 to prevent FileStream from allocating it's internal buffer
                 // 0 causes constructor to throw
                 var bufferSize = 1;
                 return new FileStream(
```

Comment:
`VirtualRazorProjectItem` can do all of this instead of taking a stream factory. (heck given the signature here it doesn't need to take any stream argument)